File: static/station.js
let io = require('socket.io-client'); //Imports the library
socket = io.connect("http://127.0.0.1:5000"); //Connects to the server


function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

socket.on('connect',async function () { //When connected
    console.log("socket connected");
    socket.emit("join", {"password":"station1", "room":"Room01"}); //This kinda shows the intention of the user with his/her details.

    while (socket.connected) {
        await sleep(4000)
        socket.emit("station", {"password":"station1", "room":"Kitchen","data":String(Math.ceil(Math.random()*100))})
        console.log("sent")
    }
});

socket.on('message',function (message){
    console.log(message)
});

socket.on('disconnect', function () {
    console.log("t")
})

File: static/station.py
import socketio, random
from time import sleep

stationId = "staTion1AB"
sio = socketio.Client()
global joined
joined = False

def join():
    """
    When asking to join a room in the server
    :return:None
    """
    global joined
    sio.emit("join", {"id":stationId})
    joined = True

def connect():
    """
    Connects the socket to the url.
    :return: None
    """
    try:
        sio.connect("wss://kingvany.pythonanywhere.com:8845")
        print("CONNECTION ESTABLISHED")
        join()
    except Exception as e:
        print(e)
        print("ERROR WHEN CONNECTING TO THE SERVER")

@sio.on("disconnect")
def disconnect():
    """
    This function is when the socket gets disconnected
    :return: None
    """
    global joined
    joined = False
    print("CONNECTION FAILED")
    connect()

if __name__ == "__main__":
    connect()
    while joined:
        print(joined)
        sleep(10)
        sio.emit("station", {"data":str(random.randrange(0,100))})